pyproject.toml

```diff
@@ -24,6 +24,7 @@ requests = "2.25.0"
 uuid = "^1.30"
 cryptography = "^3.4.8"
 tqdm = "^4.32.1"
+Shapely = "^1.8.1"
```

Comment:
CC @rad-cord 

We have so far tried to avoid adding new dependencies into our dependency tree if possible. I'll leave it up to you two to decide, but we might want  to consider adding less dependencies to clients who are using our SDK.

Reply:
Agreed. Hey Eloy. What does this dependency do and can we live without it?

@eloy-encord

Reply:
@rad-cord Shapely is a package for analysis and manipulation of geometric objects. I used it to calculate the area of the intersection of two simple polygons (as part of Intersection Over Union metric). If there is an already added package that can handle this operation I am more than happy to swap to it. On the other side, writing from scratch a code that find the intersection of two simple polygons is very troublesome and prone to errors (too many edge cases).

Reply:
@rad-cord I'll leave this decision up to you.

Reply:
Hey Eloy. I'd check out wether or not extracting just the polygon intersection code from Shapely is simple to do. If it is - I would include only the parts we need and drop shapely.
If it isn't and we have to include shapely as a dependency, can you please find the oldest version which has the intersection code? Reason being that our clients might already have shapely as a dependency in their project somewhere and we don't want to force them to upgrade it if possible.